charms/istio-pilot/src/charm.py

```diff
@@ -138,16 +138,48 @@ def handle_default_gateway(self, event):
         Side effect: self.handle_ingress() is also invoked by this handler as ingress objects
         depend on the default_gateway
         """
-        t = self.env.get_template('gateway.yaml.j2')
-        gateway = self.model.config['default-gateway']
-        manifest = t.render(name=gateway, app_name=self.app.name)
+        # Clean-up resources
         self._delete_existing_resource_objects(
             resource=self.gateway_resource,
             labels={
                 "app.juju.is/created-by": f"{self.app.name}",
                 "app.{self.app.name}.io/is-workload-entity": "true",
             },
         )
+
+        t = self.env.get_template('gateway.yaml.j2')
+        gateway = self.model.config['default-gateway']
+        secret_name = (
+            f"{self.app.name}-gateway-secret"
+            if self.model.config["ssl-crt"] and self.model.config["ssl-key"]
+            else None
+        )
+        manifest = None
+        if secret_name:
+            manifest = t.render(
+                name=gateway,
+                secret_name=secret_name,
+                ssl_crt=self.model.config["ssl-crt"],
+                ssl_key=self.model.config["ssl-key"],
+                model_name=self.model.name,
+                app_name=self.app.name,
+            )
+            secret = self.env.get_template('gateway-secret.yaml.j2')
+            manifest_secret = secret.render(
+                secret_name=secret_name,
+                ssl_crt=self.model.config["ssl-crt"],
+                ssl_key=self.model.config["ssl-key"],
+                model_name=self.model.name,
+                app_name=self.app.name,
+            )
+            self._apply_manifest(manifest_secret)
+            return
```

Comment:
We cannot `return` here as we have a call to `self.handle_ingress(event)` before the method ends. We could leave the part where we define `manifest` out of the `if secret_name:` statement. Something like this:
```python
...
if secret_name:
            secret = self.env.get_template('gateway-secret.yaml.j2')
            manifest_secret = secret.render(
                secret_name=secret_name,
                ssl_crt=self.model.config["ssl-crt"],
                ssl_key=self.model.config["ssl-key"],
                model_name=self.model.name,
                app_name=self.app.name,
            )
            self._apply_manifest(manifest_secret)
manifest = t.render(
            name=gateway,
            secret_name=secret_name,
            model_name=self.model.name,
            app_name=self.app.name,
        )
...
```